### PR TITLE
Postgres database component fixes

### LINKED ISSUE #2056: Incorrect dependencies for secure_transport_configuration
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** macOS
- **Data Safe Haven version:** develop @ 996f54f

## :package: Packages

<details>
<summary>List of packages</summary>

```none
Paste list of packages here
```

</details>

## :no_entry_sign: Describe the problem

SRE does not deploy due to incorrect dependencies for database `secure_transport_configuration`

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
  azure-native:dbforpostgresql:Configuration (sre_gitea_server_db_gitea_secure_transport_configuration):
    error: Code="ResourceNotFound" Message="The Resource 'Microsoft.DBforPostgreSQL/flexibleServers/shm-pink-sre-coral-db-server-gitea' under resource group 'shm-pink-sre-coral-rg' was not found. For more details please go to
https://aka.ms/ARMResourceNotFoundFix"
```

</details>

## :recycle: To reproduce

Attempt to deploy a fresh SRE

### PR DESCRIPTION
**Pull request by JimMadge**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :closed_umbrella: Related issues

Closes #2056

### :microscope: Tests

Tested on fresh deployment,

- Databases work (can log in _etc._, can run smoke tests)
- Teardown works without error using workaround

### CODE REVIEW

File: data_safe_haven/infrastructure/components/composite/postgresql_database.py [hunk 1]
```diff
@@ -81,9 +81,11 @@ def __init__(
             f"{self._name}_secure_transport_configuration",
             configuration_name="require_secure_transport",
             resource_group_name=props.database_resource_group_name,
-            server_name=props.database_server_name,
+            server_name=db_server.name,
             source="user-override",
             value="OFF",
+            opts=ResourceOptions.merge(child_opts, ResourceOptions(parent=db_server)),
+            tags=child_tags,
```
Comment (jemrobinson) [inline data_safe_haven/infrastructure/components/composite/postgresql_database.py]: I don't think this resource has tags - it's not a thing that exists on its own in the portal, it's a setting on an existing resource.

**Review by jemrobinson — COMMENTED** [review]
LGTM - have you tested on a fresh deployment?

**Comment by JimMadge** [comment]
Testing now :+1:

**Comment by JimMadge** [comment]
Databases are working.

Will check teardown now.

**Comment by JimMadge** [comment]
Same teardown error.

**Review by craddm — APPROVED** [review]
LGTM

**Pull request merged**